SentryTestUtils/WeakReference.swift

```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+public class WeakReference<T: AnyObject> {
```

Comment:
I don't quite understand why we need this, could you explain here and/or in a comment? Were there retain cycles in the tests that kept objects alive between test cases?

Reply:
I added a comment. Is it clear now? If not, I'm going to update the comment in a follow up PR.